android/src/main/java/com/oney/WebRTCModule/CameraCaptureController.java

```diff
@@ -85,34 +116,48 @@ public void switchCamera(SwitchCameraHandler handler) {
 
             // Nothing to switch to.
             if (deviceCount < 2) {
                 handler.onSwitchCameraDone(facingMode());
                 return;
             }
 
+            CameraVideoCapturer.CameraSwitchHandler cameraSwitchHandler = new CameraVideoCapturer.CameraSwitchHandler() {
+                @Override
+                public void onCameraSwitchDone(boolean b) {
+                    Log.d(TAG, "Invoked onCameraSwitchDone, isFrontFacing:" + b);
+                    isFrontFacing = b;
+                    handler.onSwitchCameraDone(facingMode());
+                }
+
+                @Override
+                public void onCameraSwitchError(String s) {
+                    Log.e(TAG, "Error switching camera: " + s);
+                    handler.onSwitchCameraDone(facingMode());
+                }
+            };
             // The usual case.
             if (deviceCount == 2) {
-                capturer.switchCamera(new CameraVideoCapturer.CameraSwitchHandler() {
-                    @Override
-                    public void onCameraSwitchDone(boolean b) {
-                        isFrontFacing = b;
-                        handler.onSwitchCameraDone(facingMode());
-                    }
-
-                    @Override
-                    public void onCameraSwitchError(String s) {
-                        Log.e(TAG, "Error switching camera: " + s);
-                        handler.onSwitchCameraDone(facingMode());
-                    }
-                });
-                return;
+                capturer.switchCamera(cameraSwitchHandler);
+            } else {
+                try{
+                    // We are always selecting the first one that matches the
+                    // opposite facing mode that we currently have.
+                    String deviceToSwitchTo = Arrays.stream(deviceNames).filter((deviceName) -> {
+                        Log.d(TAG, "Filtering " + deviceName + " isFrontFacing " + cameraEnumerator.isFrontFacing(deviceName));
+                        return cameraEnumerator.isFrontFacing(deviceName) != isFrontFacing;
+                    }).findFirst().get();
+                    Log.d(TAG, "Invoked switchCamera, has more than 02 cameras, will switch to device:" + deviceToSwitchTo);
+                    capturer.switchCamera(cameraSwitchHandler, deviceToSwitchTo);
+                }catch (Exception e) {
+                    Log.d(TAG, "Failed to switch the camera device:" + e.getMessage());
+                }
+                // If we are here the device has more than 2 cameras. Cycle through them
+                // and switch to the first one of the desired facing mode.
+                // switchCamera(!isFrontFacing, deviceCount, handler);
             }
-
-            // If we are here the device has more than 2 cameras. Cycle through them
-            // and switch to the first one of the desired facing mode.
-            switchCamera(!isFrontFacing, deviceCount, handler);
```

Comment:
is this `switchCamera()` helper method now dead code?

Reply:
wait, why again did this method not work? did it never result in a switch to the desired facing camera?

Reply:
Yep, this is now dead code, but since It was from the upstream, I thought If we didn't remove It, maybe could make things easier when bringing the changes from upstream.

Reply:
It was not working because Samsung had 04 cameras. Just so It is easy to understand, let's call them:
- FrontCam01, FrontCam02, RearCam01, RearCam02.

With our current logic in `daily-js`, when we invoke `setCamera`, we check if our current camera facing mode is different from the facing mode from the camera that the user has asked. If yes, we invoke `switchCamera`.

You can see this logic here(https://github.com/daily-co/pluot-core/blob/8c07487198ba85738c8a50240f8f1cbff8f55ad0/call-machine/EmbeddedUtil.js#L2342).

The issue is if the user requested for a front-facing camera for instance, one time we would get the FrontCam01 and another time we would get the FrontCam02. Instead of always using the same camera.